core/adapters/random.go

```diff
@@ -56,33 +69,75 @@ func (ra *Random) TaskType() models.TaskType {
 
 // Perform returns the the proof for the VRF output given seed, or an error.
 func (ra *Random) Perform(input models.RunInput, store *store.Store) models.RunOutput {
-	key, err := getKey(ra, input)
+	key, preSeed, blockHash, blockNum, err := getInputs(ra, input, store)
 	if err != nil {
-		return models.NewRunOutputError(errors.Wrapf(err, "bad key for vrf task"))
-	}
-	seed, err := getSeed(input)
-	if err != nil {
-		return models.NewRunOutputError(errors.Wrap(err, "bad seed for vrf task"))
+		return models.NewRunOutputError(err)
 	}
-	solidityProof, err := store.VRFKeyStore.GenerateProof(key, seed)
+	solidityProof, err := store.VRFKeyStore.GenerateProof(key, preSeed,
+		*blockHash, *blockNum)
 	if err != nil {
 		return models.NewRunOutputError(err)
 	}
 	ethereumByteArray := fmt.Sprintf("0x%x", utils.EVMEncodeBytes(solidityProof[:]))
 	return models.NewRunOutputCompleteWithResult(ethereumByteArray)
 }
 
-// getSeed returns the numeric seed for the vrf task, or an error
-func getSeed(input models.RunInput) (*big.Int, error) {
+// getInputs parses the JSON input and returns an error, or the values in it.
```

Comment:
If we're going to have a comment, can it come with a bit of useful context?

Reply:
```golang
// getInputs parses the JSON input for the values needed by the random adapter,
// or returns an error.
```